src/udev/scsi_id/scsi_serial.c

```diff
@@ -182,19 +183,21 @@ static int scsi_dump_sense(struct scsi_id_device *dev_scsi,
                  */
                 s = sense_buffer[7] + 8;
                 if (sb_len < s) {
-                        log_debug("%s: sense buffer too small %d bytes, %d bytes too short",
-                                  dev_scsi->kernel, sb_len, s - sb_len);
-                        return -1;
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "%s: sense buffer too small %d bytes, %d bytes too short",
+                                               dev_scsi->kernel, sb_len,
+                                               s - sb_len);
                 }
                 if (IN_SET(code, 0x0, 0x1)) {
                         sense_key = sense_buffer[2] & 0xf;
                         if (s < 14) {
                                 /*
                                  * Possible?
                                  */
-                                log_debug("%s: sense result too" " small %d bytes",
-                                          dev_scsi->kernel, s);
-                                return -1;
+                                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "%s: sense result too"
+                                                       " small %d bytes",
```

Comment:
Please don't split strings like that. (My solution is to patch `spatch` to bump the string length limit to something very large and use that binary instead... But since it's just one case here, it should be enough to patch it by hand ;)) ).

Reply:
I somehow managed to completely miss this one, thanks for noticing!